app/Jobs/ProcessMediaWikiJobsJob.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace App\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Log;
+use Maclof\Kubernetes\Client;
+use Maclof\Kubernetes\Models\Job as KubernetesJob;
+
+class ProcessMediaWikiJobsJob implements ShouldQueue, ShouldBeUnique
+{
+    use InteractsWithQueue, Queueable;
+
+    private string $wikiDomain;
+    private const API_JOB_CONCURRENCY_LIMIT = intval(getenv('API_JOB_CONCURRENCY_LIMIT') || '8', 10);
+
+    public function __construct (string $wikiDomain)
+    {
+        $this->wikiDomain = $wikiDomain;
+    }
+
+    public function uniqueId(): string
+    {
+        return $this->wikiDomain;
+    }
+
+    public function handle (Client $kubernetesClient): void
+    {
+        $kubernetesClient->setNamespace('api-jobs');
+
+        $numJobs = $kubernetesClient->jobs()->setFieldSelector([
+            'status.phase' => 'Running',
```

Comment:
Theoretically, this could also be `Pending`, but I don't have a clear idea how to query for jobs that are either `Running` or `Pending` right now?

Reply:
It seems it doesn't even work at all like this and we'll have to manually filter out the result sets manually ourselves.